### ISSUE TITLE
Plugin crashes when using autocomplete shortcut while method signature popover is active

### ISSUE THREAD

**Issue by mochtu** [Microsoft/TypeScript-Sublime-Plugin#379]
Sublime Text 3 / OSX

For some reason, when I accidentally use the Ctrl+Space (autocomplete) shortcut while a method signature popover is displayed, the typescript plugin stops working (without an error).

An example would be:
```
angular.module(∆)
```
where "∆" represents the cursor position.

When I press Ctrl+Space again, I get the following stack trace in the sublime console:

```
insert_snippet
Exception in thread Thread-1:
Traceback (most recent call last):
  File "./threading.py", line 901, in _bootstrap_inner
  File "./threading.py", line 858, in run
  File "/Users/mo/Library/Application Support/Sublime Text 3/Packages/TypeScript/typescript/libs/node_client.py", line 283, in __reader
    if NodeCommClient.read_msg(stream, msgq, eventq, asyncReq, proc, eventHandlers):
  File "/Users/mo/Library/Application Support/Sublime Text 3/Packages/TypeScript/typescript/libs/node_client.py", line 182, in read_msg
    data_dict = json_helpers.decode(data_json)
  File "/Users/mo/Library/Application Support/Sublime Text 3/Packages/TypeScript/typescript/libs/json_helpers.py", line 18, in decode
    return json.loads(json_str)
  File "./json/__init__.py", line 319, in loads
  File "./json/decoder.py", line 352, in decode
  File "./json/decoder.py", line 368, in raw_decode
  File "./json/scanner.py", line 38, in _scan_once
  File "./json/decoder.py", line 198, in JSONObject
  File "./json/scanner.py", line 40, in _scan_once
  File "./json/decoder.py", line 242, in JSONArray
  File "./json/scanner.py", line 38, in _scan_once
  File "./json/decoder.py", line 198, in JSONObject
  File "./json/scanner.py", line 35, in _scan_once
  File "./json/decoder.py", line 100, in py_scanstring
ValueError: Invalid control character '\r' at: line 1 column 65532 (char 65531)
```

**Comment by zhengbli** [comment]
What version of node are you using?

**Comment by mochtu** [comment]
~ $ node -v
v4.2.1
~ $ npm -v
2.14.7

**Comment by mochtu** [comment]
And I checked, node_path is correctly set to that executable.

**Comment by zhengbli** [comment]
Thanks @mochtu. This is a known issue that `fs.writeSync` truncates long response messages from the server under Node v4 on OSX. I'm working on a fix right now.

**Comment by mochtu** [comment]
Ah, great to hear. Couldn't find any reference to that known issue. Looking forward to your fix :-)

**Comment by zhengbli** [comment]
Hi @mochtu, a recent change should have fixed the issue. Could you try the bit in the master branch? Thanks!

**Comment by mochtu** [comment]
The issue is fixed for me with #383 ! Thanks
